### PR TITLE
Minor optimizations and refactors

### PR DESCRIPTION
**Pull request by alanssitis**
Did not touch driver_halleffect cuz its kinda nasty and I don't have
tuples in this language :(

### CODE REVIEW

File: firmware-nucleo/smart-air-hockey-table/App/driver_display.c [hunk 1]
```diff
@@ -104,28 +105,29 @@ void Driver_Display_Print(Display display, uint_fast8_t line, uint_fast8_t offse
...
-		uint8_t character = *(string++);
+		uint8_t character = *(curr++);
 		if (character == '\t')
 		{
 			inverted = !inverted;
 			continue;
 		}
-		if (character < FONT_6X8_MIN || character > FONT_6X8_MAX) character = ' ';
+		if (character < FONT_6X8_MIN || character > FONT_6X8_MAX)
+			character = ' ';
```
Comment (tream2001) [inline firmware-nucleo/smart-air-hockey-table/App/driver_display.c:130]: I know it's not technically required, but,  if we are putting the instruction onto a new line, then shouldn't we add curly braces as well?

Comment (alanssitis) [reply firmware-nucleo/smart-air-hockey-table/App/driver_display.c:131]: done

File: firmware-nucleo/smart-air-hockey-table/App/driver_display.c [hunk 2]
```diff
@@ -140,8 +144,17 @@ void Driver_Display_Print(Display display, uint_fast8_t line, uint_fast8_t offse
 	transmit_end();
 }
 
+static inline void transmit_large_word(const uint8_t font_char[])
+{
+	for (uint_fast32_t i = 0; i < FONT_ARRAY_SIZE; i++)
+	{
+		transmit_word(font_char[i]);
```
Comment (WDobert) [inline firmware-nucleo/smart-air-hockey-table/App/driver_display.c:151]:
```suggestion
		transmit_word(char_data[i]);
```

**Review by WDobert — CHANGES_REQUESTED** [review]
Just changing some names to avoid confusion, everything else looks great.

**Review by WDobert — APPROVED** [review]

**Pull request merged**